packages/tupaia-web-server/src/routes/MapOverlaysRoute.ts

```diff
@@ -128,19 +129,28 @@ export class MapOverlaysRoute extends Route<MapOverlaysRequest> {
       (group: MapOverlayGroup) => group.code === ROOT_MAP_OVERLAY_CODE,
     );
 
-    const nestedGroups = nestOverlayGroups(
-      relationsByParentId,
-      groupsById,
-      overlaysById,
-      rootOverlayGroup,
-    );
+    if (rootOverlayGroup) {
+      const nestedGroups = nestOverlayGroups(
+        relationsByParentId,
+        groupsById,
+        overlaysById,
+        rootOverlayGroup,
+      );
```

Comment:
```suggestion
      const nestedGroups = rootOverlayGroup && nestOverlayGroups(
        relationsByParentId,
        groupsById,
        overlaysById,
        rootOverlayGroup,
      );
```

These last ones are non-essential. It'd be a little cleaner to not re-write that return case